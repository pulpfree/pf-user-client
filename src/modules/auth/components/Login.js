import React, { Component } from 'react'
// import PropTypes from 'prop-types'

import { Field, reduxForm } from 'redux-form'
import { Paper } from 'react-md'

class Login extends Component {

  _onHandleSubmit(values) {
    console.log('values in _onHandleSubmit:', values)
  }

  render() {

    return (
      <div>Login Component
        <Paper>

        </Paper>
      </div>
    )
  }
}

Login = reduxForm({ // eslint-disable-line no-class-assign
  form:           'login-form',
  // validate,
})(Login)

export default Login