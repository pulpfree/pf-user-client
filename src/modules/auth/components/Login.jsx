import React, { Component } from 'react'


import g from 'glamorous'
import {
  Paper,
  Toolbar,
} from 'react-md'

import Header from '../../common/components/Header'
import LoginForm from '../containers/LoginForm'

class Login extends Component {

  render() {

    return (
      <div>
        <Header />
        <g.Div
          className="md-cell md-cell--6"
          margin={'auto'}
          marginTop={'30px'}
        >
          <Paper>
            <Toolbar
              colored
              title="Enter Login Credentials"
            />
            <LoginForm />
          </Paper>
        </g.Div>
      </div>
    )
  }
}

export default Login
