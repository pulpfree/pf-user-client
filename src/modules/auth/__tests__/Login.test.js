/* eslint no-undef: "off" */
import React from 'react'

import { mount } from 'enzyme'

import Login from '../components/Login'

describe('Login', () => {
  let props
  let mountedC
  const loginC = () => {
    if (!mountedC) {
      mountedC = mount(
        <Login {...props} />
      )
    }
    return mountedC
  }

  beforeEach(() => {
    mountedC = undefined
  })

  it('always renders a component', () => {
    expect(loginC().find('div').length).toBeGreaterThan(0)
  })

  /*it('matches snapshot', () => {
    expect(loginC()).toMatchSnapshot()
  })*/

})