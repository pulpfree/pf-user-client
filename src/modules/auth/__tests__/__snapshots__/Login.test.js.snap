// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <glamorous.Div
          className="md-cell md-cell--6"
          margin="auto"
          marginTop="30px"
>
          <Paper
                    component="div"
                    zDepth={1}
          >
                    <Toolbar
                              colored={true}
                              component="header"
                              singleColor={true}
                              title="Enter Login Credentials"
                    />
                    <Connect(ReduxForm) />
          </Paper>
</glamorous.Div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Paper
            component="div"
            zDepth={1}
>
            <Toolbar
                        colored={true}
                        component="header"
                        singleColor={true}
                        title="Enter Login Credentials"
            />
            <Connect(ReduxForm) />
</Paper>,
          "className": "md-cell md-cell--6",
          "margin": "auto",
          "marginTop": "30px",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Toolbar
                colored={true}
                component="header"
                singleColor={true}
                title="Enter Login Credentials"
/>,
              <Connect(ReduxForm) />,
            ],
            "component": "div",
            "zDepth": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "colored": true,
                "component": "header",
                "singleColor": true,
                "title": "Enter Login Credentials",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <glamorous.Div
            className="md-cell md-cell--6"
            margin="auto"
            marginTop="30px"
>
            <Paper
                        component="div"
                        zDepth={1}
            >
                        <Toolbar
                                    colored={true}
                                    component="header"
                                    singleColor={true}
                                    title="Enter Login Credentials"
                        />
                        <Connect(ReduxForm) />
            </Paper>
</glamorous.Div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Paper
              component="div"
              zDepth={1}
>
              <Toolbar
                            colored={true}
                            component="header"
                            singleColor={true}
                            title="Enter Login Credentials"
              />
              <Connect(ReduxForm) />
</Paper>,
            "className": "md-cell md-cell--6",
            "margin": "auto",
            "marginTop": "30px",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Toolbar
                  colored={true}
                  component="header"
                  singleColor={true}
                  title="Enter Login Credentials"
/>,
                <Connect(ReduxForm) />,
              ],
              "component": "div",
              "zDepth": 1,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "colored": true,
                  "component": "header",
                  "singleColor": true,
                  "title": "Enter Login Credentials",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
